core/src/main/java/org/apache/accumulo/core/client/admin/TableOperations.java

```diff
@@ -1014,39 +1014,15 @@ default TimeType getTimeType(String tableName) throws TableNotFoundException {
   }
 
   /**
-   * Initiates setting a table to an on-demand state, but does not wait for action to complete
+   * Sets the hosting goal for a range of Tablets in the specified table.
    *
-   * @param tableName the table to set to an on-demand state
-   * @throws AccumuloException when there is a general accumulo error
-   * @throws AccumuloSecurityException when the user does not have the proper permissions
-   * @since 3.1.0
-   */
-  void onDemand(String tableName)
-      throws AccumuloSecurityException, AccumuloException, TableNotFoundException;
-
-  /**
-   * Initiates setting a table to an on-demand state, optionally waits for action to complete
-   *
-   * @param tableName the table to set to an on-demand state
-   * @param wait if true, then will not return until table state is set to an on-demand state
-   * @throws AccumuloException when there is a general accumulo error
-   * @throws AccumuloSecurityException when the user does not have the proper permissions
-   * @since 3.1.0
-   */
-  void onDemand(String tableName, boolean wait)
-      throws AccumuloSecurityException, AccumuloException, TableNotFoundException;
-
-  /**
-   * Check if a table is on-demand through its current goal state only. Could run into issues if the
-   * current state of the table is in between states. If you require a specific state, call
-   * <code>onDemand(tableName, true)</code>, this will wait until the table reaches the desired
-   * state before proceeding.
-   *
-   * @param tableName the table to check if in an on-demand state
-   * @throws AccumuloException when there is a general accumulo error
-   * @return true if table's goal state is online
-   * @since 3.1.0
+   * @param tableName table name
+   * @param range tablet range
+   * @param goal hosting goal (one of {@literal ALWAYS, DEFAULT, NEVER})
+   * @since ELASTICITY_TODO
    */
-  boolean isOnDemand(String tableName) throws AccumuloException, TableNotFoundException;
-
+  default void setTabletHostingGoal(String tableName, Range range, String goal)
```

Comment:
The goal parameter should probably be an enum.  Could use the TabletHostingGoal minus the static methods, maybe have a TabletHostingGoalImpl class that has the static methods.

Reply:
Updated based on your suggestion in d579d7b